File: sdkwork-backend-sdk-go/api/shop.go
package api

import (
    "fmt"
    sdktypes "github.com/sdkwork/backend-sdk/types"
    sdkhttp "github.com/sdkwork/backend-sdk/http"
)

type ShopApi struct {
    client *sdkhttp.Client
}

func NewShopApi(client *sdkhttp.Client) *ShopApi {
    return &ShopApi{client: client}
}

// Update an existing shop
func (a *ShopApi) Update(body sdktypes.PlusShopForm) (sdktypes.PlusApiResultPlusShopVO, error) {
    raw, err := a.client.Put(BackendApiPath("/shop"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPlusShopVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusShopVO](raw)
}

// Create a new shop
func (a *ShopApi) Create(body sdktypes.PlusShopForm) (sdktypes.PlusApiResultPlusShopVO, error) {
    raw, err := a.client.Post(BackendApiPath("/shop"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPlusShopVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusShopVO](raw)
}

// Get shops by page
func (a *ShopApi) ListByPage(body *sdktypes.QueryListForm, query map[string]interface{}) (sdktypes.PlusApiResultPagePlusShopVO, error) {
    raw, err := a.client.Post(BackendApiPath("/shop/list"), body, query, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPagePlusShopVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPagePlusShopVO](raw)
}

// Get all shops
func (a *ShopApi) ListAllEntities(body *sdktypes.QueryListForm) (sdktypes.PlusApiResultListPlusShopVO, error) {
    raw, err := a.client.Post(BackendApiPath("/shop/list/all"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultListPlusShopVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListPlusShopVO](raw)
}

// Get a shop by ID
func (a *ShopApi) GetById(id string) (sdktypes.PlusApiResultPlusShopVO, error) {
    raw, err := a.client.Get(BackendApiPath(fmt.Sprintf("/shop/%s", id)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultPlusShopVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusShopVO](raw)
}

// Delete a shop
func (a *ShopApi) Delete(id string) (sdktypes.PlusApiResultBoolean, error) {
    raw, err := a.client.Delete(BackendApiPath(fmt.Sprintf("/shop/%s", id)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultBoolean
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultBoolean](raw)
}

File: sdkwork-backend-sdk-typescript/src/api/shop.ts
import { backendApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { PlusApiResultBoolean, PlusApiResultListPlusShopVO, PlusApiResultPagePlusShopVO, PlusApiResultPageResultShopBrandVO, PlusApiResultPageResultShopCsVO, PlusApiResultPageResultShopStaffVO, PlusApiResultPlusShopVO, PlusApiResultShopBrandVO, PlusApiResultShopCsVO, PlusApiResultShopDecorationVO, PlusApiResultShopStaffVO, PlusShopForm, QueryListForm, ShopBrandRequest, ShopCsRequest, ShopDecorationRequest, ShopListRequest, ShopStaffRequest } from '../types';


export class ShopApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** Update an existing shop */
  async update(body: PlusShopForm): Promise<PlusApiResultPlusShopVO> {
    return this.client.put<PlusApiResultPlusShopVO>(backendApiPath(`/shop`), body);
  }

/** Create a new shop */
  async create(body: PlusShopForm): Promise<PlusApiResultPlusShopVO> {
    return this.client.post<PlusApiResultPlusShopVO>(backendApiPath(`/shop`), body);
  }

/** Update shop staff */
  async updateStaff(body: ShopStaffRequest): Promise<PlusApiResultShopStaffVO> {
    return this.client.put<PlusApiResultShopStaffVO>(backendApiPath(`/shop/admin/staff`), body);
  }

/** Create shop staff */
  async createStaff(body: ShopStaffRequest): Promise<PlusApiResultShopStaffVO> {
    return this.client.post<PlusApiResultShopStaffVO>(backendApiPath(`/shop/admin/staff`), body);
  }

/** Update shop customer service */
  async updateCustomerService(body: ShopCsRequest): Promise<PlusApiResultShopCsVO> {
    return this.client.put<PlusApiResultShopCsVO>(backendApiPath(`/shop/admin/cs`), body);
  }

/** Create shop customer service */
  async createCustomerService(body: ShopCsRequest): Promise<PlusApiResultShopCsVO> {
    return this.client.post<PlusApiResultShopCsVO>(backendApiPath(`/shop/admin/cs`), body);
  }

/** Update shop brand */
  async updateBrand(body: ShopBrandRequest): Promise<PlusApiResultShopBrandVO> {
    return this.client.put<PlusApiResultShopBrandVO>(backendApiPath(`/shop/admin/brand`), body);
  }

/** Create shop brand */
  async createBrand(body: ShopBrandRequest): Promise<PlusApiResultShopBrandVO> {
    return this.client.post<PlusApiResultShopBrandVO>(backendApiPath(`/shop/admin/brand`), body);
  }

/** Get shops by page */
  async listByPage(body?: QueryListForm, params?: QueryParams): Promise<PlusApiResultPagePlusShopVO> {
    return this.client.post<PlusApiResultPagePlusShopVO>(backendApiPath(`/shop/list`), body, params);
  }

/** Get all shops */
  async listAllEntities(body?: QueryListForm): Promise<PlusApiResultListPlusShopVO> {
    return this.client.post<PlusApiResultListPlusShopVO>(backendApiPath(`/shop/list/all`), body);
  }

/** Shop staff page */
  async listStaff(body?: ShopListRequest): Promise<PlusApiResultPageResultShopStaffVO> {
    return this.client.post<PlusApiResultPageResultShopStaffVO>(backendApiPath(`/shop/admin/staff/list`), body);
  }

/** Get shop decoration */
  async getDecoration(): Promise<PlusApiResultShopDecorationVO> {
    return this.client.get<PlusApiResultShopDecorationVO>(backendApiPath(`/shop/admin/decoration`));
  }

/** Save shop decoration */
  async saveDecoration(body: ShopDecorationRequest): Promise<PlusApiResultShopDecorationVO> {
    return this.client.post<PlusApiResultShopDecorationVO>(backendApiPath(`/shop/admin/decoration`), body);
  }

/** Shop customer service page */
  async listCustomerService(body?: ShopListRequest): Promise<PlusApiResultPageResultShopCsVO> {
    return this.client.post<PlusApiResultPageResultShopCsVO>(backendApiPath(`/shop/admin/cs/list`), body);
  }

/** Shop brand page */
  async listBrands(body?: ShopListRequest): Promise<PlusApiResultPageResultShopBrandVO> {
    return this.client.post<PlusApiResultPageResultShopBrandVO>(backendApiPath(`/shop/admin/brand/list`), body);
  }

/** Get a shop by ID */
  async getById(id: string | number): Promise<PlusApiResultPlusShopVO> {
    return this.client.get<PlusApiResultPlusShopVO>(backendApiPath(`/shop/${id}`));
  }

/** Delete a shop */
  async delete(id: string | number): Promise<PlusApiResultBoolean> {
    return this.client.delete<PlusApiResultBoolean>(backendApiPath(`/shop/${id}`));
  }

/** Delete shop staff */
  async deleteStaff(id: string | number): Promise<PlusApiResultBoolean> {
    return this.client.delete<PlusApiResultBoolean>(backendApiPath(`/shop/admin/staff/${id}`));
  }

/** Delete shop customer service */
  async deleteCustomerService(id: string | number): Promise<PlusApiResultBoolean> {
    return this.client.delete<PlusApiResultBoolean>(backendApiPath(`/shop/admin/cs/${id}`));
  }

/** Delete shop brand */
  async deleteBrand(id: string | number): Promise<PlusApiResultBoolean> {
    return this.client.delete<PlusApiResultBoolean>(backendApiPath(`/shop/admin/brand/${id}`));
  }
}

export function createShopApi(client: HttpClient): ShopApi {
  return new ShopApi(client);
}
